game-core/src/main/java/games/strategy/engine/framework/map/file/system/loader/DownloadedMaps.java

```diff
@@ -65,4 +68,51 @@ public boolean hasGame(final String gameName) {
         .findAny()
         .map(DefaultGameChooserEntry::getUri);
   }
+
+  /**
+   * For a given a map name (fuzzy matched), if the map is installed, will return the map version
+   * installed othewrise returns an empty. Fuzzy matching means we will do name normalization and
+   * replace spaces with underscores, convert to lower case, etc.
+   */
+  public static Optional<Integer> getMapVersionByName(final String mapName) {
+    final String normalizedMapName = normalizeName(mapName);
+
+    // see if we have a map folder that matches the target name
+    // if so, check for a .properties file
+    // if that exists, then return the map version value from the properties file
+    return FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder()).stream()
+        .filter(file -> normalizeName(file.getName()).equals(normalizedMapName))
+        .findAny()
+        .map(
+            file ->
+                ClientFileSystemHelper.getUserMapsFolder()
+                    .toPath()
+                    .resolve(file.getName() + ".properties")
+                    .toFile())
+        .filter(File::exists)
+        .flatMap(DownloadedMaps::readVersionFromPropertyFile);
+  }
+
+  private static Optional<Integer> readVersionFromPropertyFile(final File propertyFile) {
+    return DownloadFileProperties.loadForZipPropertyFile(propertyFile).getVersion();
+  }
+
+  /**
+   * Returns a normalized version of the input. Trims off a '.properties' suffix if present,
+   * converts to lower case and replaces all spaces with underscores.
+   */
+  private static String normalizeName(final String inputName) {
+    String normalizedName = inputName;
+    if (inputName.endsWith(".zip.properties")) {
+      normalizedName = inputName.substring(0, inputName.indexOf(".zip.properties"));
+    }
+    if (normalizedName.endsWith("-master")) {
+      normalizedName = inputName.substring(0, inputName.indexOf("-master"));
+    }
+
+    normalizedName = normalizedName.replaceAll(" ", "_");
+    normalizedName = normalizedName.toLowerCase();
+
+    return normalizedName;
```

Comment:
In my opinion using a RegEx simplifies the code by a reasonable amount
```java
return normalizedName.toLowerCase()
  .replaceAll(" ", "_")
  .replaceAll("(?:-master)?(?:.zip.properties)?$", "");
```

Reply:
That is more concise, my 2 cents it's kinda perl-esque in that it's a lot shorter, but more difficult to understand. I'd also start to wonder about the runtime as we do this normalization n-squared times.

To another extent, this code I think will go away when we start matching map name from a `map.yml` instead of deriving it from folder name. Let's hold off on changes to how that is done as it could be churn